sandbox/rocky/tf/algos/ddpg/ddpg.py

```diff
@@ -0,0 +1,233 @@
+        self._actor_net.set_grad(self._critic_net.action_grads)
+
+        #setup network to target network params update op
+        self._actor_net.setup_target_net(self._target_actor, self._tau)
+        self._critic_net.setup_target_net(self._target_critic, self._tau)
+
+        self._global_step = tf.Variable(
+            initial_value=0, name='global_step', trainable=False)
+
+        self.load_session()
+
+    def _train_net(self):
+        # Get a batch.
+        [state, action, reward, terminal,
+         next_state] = self._replay_buffer.get_batch_data(self._batch_size)
+        reward = reward.reshape(-1, 1)
+        terminal = terminal.reshape(-1, 1)
+
+        target_action = self._target_actor.predict(next_state)
+        target_Q = self._target_critic.predict_target_Q(
+            next_state, target_action, reward, terminal)
+
+        self._critic_net.train(state, action, target_Q)
+        self._actor_net.train(state)
+
+        self._actor_net.update_target_net()
+        self._critic_net.update_target_net()
+        return
+
+    def _report_total_reward(self, reward, step):
+        summary = tf.Summary()
+        summary.value.add(tag='rollout/reward', simple_value=float(reward))
+        summary.value.add(
+            tag='train/episode_reward', simple_value=float(reward))
+        if self._summary_writer:
+            self._summary_writer.add_summary(summary, step)
+
+    def predict(self, state):
+        action = self._actor_net.predict(np.array(state).reshape(1, -1))[0]
+        if self._action_noise:
+            noise = self._action_noise.gen()
+            action = action + noise
+        action = np.clip(action, self._action_range[0], self._action_range[1])
+        return action
+
+    def train(self,
+              epochs=500,
+              epoch_cycles=20,
+              rollout_steps=100,
+              train_steps=50):
+        state = self._env.reset()
+        if self._action_noise:
+            self._action_noise.reset()
+        total_reward = 0.0
+        episode_step = self._session.run(self._global_step)
+
+        for epoch in range(epochs):
+            for step in range(epoch_cycles):
+                for rollout in range(rollout_steps):
+                    if self._plot:
+                        self._env.render()
+                    action = self.predict(state)
+
+                    next_state, reward, terminal, info = self._env.step(
+                        action * self._max_action)
+                    self._replay_buffer.add_data(state, action,
+                                                 reward * self._reward_scale,
+                                                 terminal, next_state)
+                    state = next_state
+                    total_reward += reward
+                    if terminal:
+                        self._report_total_reward(total_reward, episode_step)
+                        print("epoch %d, total reward %lf\n" % (episode_step,
```

Comment:
Use rllab's logger for this (`logger.record_tabular('key', value)`) and log more performance measures than reward, e.g.:
- actor loss
- critic loss
- max and average predicted Q value
- etc. (maybe you can think of more things that could help us understand the performance)